enumerable_methods.rb

```diff
@@ -0,0 +1,104 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    i = 0
+    while i < length
+      yield(self[i])
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    i = 0
+    while i < length
+      yield(self[i], i)
+      i += 1
+    end
+    self
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    i = 0
+    result = []
+    while i < length
+      answer = yield(self[i])
+      result << self[i] unless answer == false
+      i += 1
+    end
+    result
+  end
+
+  def my_all?(classification = nil)
+    test = true
+    checked = classification.class
+    my_each { |item| return false unless yield[item] == false } if block_given?
+    my_each { |item| return false unless item.class == classification } if checked == Class
+    my_each { |item| return false unless item =~ classification } if checked == Regexp
+    my_each { |item| return false unless item == classification } if [Integer, String].include?(checked)
+    my_each { |item| return false unless item } if !classification && !block_given?
+    test
+  end
+
+  def my_any?(classification = nil)
+    test = false
+    my_each { |item| return true unless yield(item) == false } if block_given?
+    my_each { |item| return true unless item.class == classification } if classification.class == Class
+    my_each { |item| return true unless item =~ classification } if classification.class == Regexp
+    my_each { |item| return true unless item == classification } if [Integer, String].include?(classification.class)
+    my_each { |item| return true if item } if !classification && !block_given?
+    test
+  end
+
+  def my_none?(classification = nil)
+    test = true
+    my_each { |item| return false if yield(item) == true } if block_given?
+    my_each { |item| return false if item.class == classification } if classification.class == Class
+    my_each { |item| return false if item =~ classification } if classification.class == Regexp
+    my_each { |item| return false if item == classification } if [Integer, String].include?(classification.class)
+    my_each { |item| return false if item } if !classification && !block_given?
+    test
+  end
+
+  def my_count(classification = nil)
+    counter = 0
+    my_each { |item| counter += 1 if item == classification } if [Integer, String].include?(classification.class)
+    my_each { |item| counter += 1 if yield(item) == true } if [Integer, String].include?(classification.class)
+    my_each { |item| counter += 1 if item.class == classification } if classification.class == Class
+    return length if !classification && !block_given?
+
+    counter
+  end
```

Comment:
**[REQUIRED]** 
* Counts the number of items in enum that are equal to item if an argument is given.
```
Failure/Error: expect(array.my_count(LOWEST_VALUE)).to eq array.count(LOWEST_VALUE)
     
     LocalJumpError:
       no block given (yield)
     # ./enumerable_methods.rb:73:in `block in my_count'
     # ./enumerable_methods.rb:9:in `my_each'
     # ./enumerable_methods.rb:73:in `my_count'
     # ./spec/enumerables_spec.rb:187:in `block (3 levels) in <top (required)>'
```
* Counts the number of elements yielding a true value if a block is given.
```
Failure/Error: expect(array.my_count(&block)).to eq array.count(&block)
     
       expected: 54
            got: 0
     
       (compared using ==)
     # ./spec/enumerables_spec.rb:191:in `block (3 levels) in <top (required)>'
```